.github/workflows/test-helm-samples.yml

```diff
@@ -69,8 +75,12 @@ jobs:
 
       - name: Install Helm chart for project-v4-with-plugins
         run: |
-          helm install my-release testdata/project-v4-with-plugins/dist/chart --create-namespace --namespace project-v4-with-plugins-system
+          helm install my-release testdata/project-v4-with-plugins/dist/chart --create-namespace --namespace project-v4-with-plugins-system --set prometheus.enable=true
 
       - name: Check Helm release status
         run: |
           helm status my-release --namespace project-v4-with-plugins-system
+
+      - name: Check Presence of ServiceMonitor
+        run: |
+          kubectl wait --namespace project-v4-with-plugins-system --for=jsonpath='{.kind}'=ServiceMonitor servicemonitor/project-v4-with-plugins-controller-manager-metrics-monitor
```

Comment:
Fails with exit 1 code if the resource does not exist.

Reply:
I think we have a bug.
For we are able to check it out we need enable prometheus in the HelmChart value.

By default it is

```
prometheus:
  enable: false
```

we need to use shell to change it in the GitHUb action to set true
Something like example: https://github.com/kubernetes-sigs/kubebuilder/blob/master/.github/workflows/test-e2e-samples.yml#L43-L44

Then the error will be faced
We might either to change : 
pkg/plugins/optional/helm/v1alpha/scaffolds/internal/templates/github/test_chart.go

OR the chart

Reply:
I'm passing the value to the helm install command in line 78 :)
Check the --set flag of the [helm install](https://helm.sh/docs/helm/helm_install/)

Reply:
nit: how about adding an explicit timeout flag for `kubectl wait`:
```
--timeout=60s
```

Also, `--for=''` could be another choice to consider?

Reply:
This is a bit of an hack since the kubectl wait is supposed to be used with resources that have a Status field.
The command fails right away if the resource doesn't exist so the `--timeout` flag isn't really needed here (I guess `--timeout=0` would be ok).

If `kubectl get` on a non existent resource would somehow return a non zero exit code that's what I would have used but I don't think that's possible. And between processing the output of `kubectl get` and simply using the hack above, I think the hack above is easier to understand.


The `--for=''` could be an option yea; my thought was that it could lead to confusion as it might be less clear.